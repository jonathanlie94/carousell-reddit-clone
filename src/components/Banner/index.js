import React, { Component } from 'react';
import styled from 'styled-components';
import { theme } from 'ui';

const StyledDiv = styled.div`
  background-color: ${theme.colors.primary};
  color: ${theme.colors.accent};
  display: flex;
`;

const StyledImg = styled.img`
  width: 100%;
  height: 200px;
`;

class Banner extends Component {
  render() {
    return (
      <StyledDiv>
        <StyledImg src="logo.svg" />
      </StyledDiv>
    );
  }
}

export default Banner;
