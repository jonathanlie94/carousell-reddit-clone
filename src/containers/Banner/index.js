import React, { Component } from 'react';
import styled from 'styled-components';
import { theme } from 'ui';

const StyledDiv = styled.div`
  background-color: ${theme.colors.primary};
  color: ${theme.colors.accent};
  display: flex;
`;

class Banner extends Component {
  render() {
    return <StyledDiv>Banner</StyledDiv>;
  }
}

export default Banner;
