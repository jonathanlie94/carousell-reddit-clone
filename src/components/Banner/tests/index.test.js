import React from 'react';
import { shallow } from 'enzyme';
import Banner from '../';

describe('<Banner />', () => {
  it('should render children when passed in', () => {
    const renderedComponent = shallow(
      <Banner>
        <div className="unique-class-name" />
      </Banner>
    );
    expect(
      renderedComponent.contains(<div className="unique-class-name" />)
    ).toEqual(true);
  });
});
