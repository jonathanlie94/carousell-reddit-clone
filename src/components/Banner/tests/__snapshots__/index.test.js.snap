// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should match previous snapshot 1`] = `
.dKaCtQ {
  background-color: #00C3FF;
  background-image: url(/banner.jpg);
  background-size: cover;
  color: #C8C8C8;
  width: 100%;
  height: 250px;
  position: relative;
}

.fPPTTC {
  background-color: transparent;
  font-family: Overwatch;
  font-style: italic;
  text-decoration: none;
  text-shadow: 0 1px 1px;
  color: #C8C8C8;
  font-size: 6rem;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  margin-left: 2rem;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.dEhpst {
  color: #666;
}

@media (max-width: 480px) {
  .fPPTTC {
    font-size: 4rem;
  }
}

<div
  className="sc-bdVaJa dKaCtQ"
>
  <styled.span>
    <span
      className="sc-bwzfXH fPPTTC"
    >
      REDDIT
      <styled.span>
        <span
          className="sc-htpNat dEhpst"
        >
          CLONE
        </span>
      </styled.span>
    </span>
  </styled.span>
</div>
`;
